shop/src/main/java/ro/msg/learning/shop/controller/CategoryController.java

```diff
@@ -0,0 +1,68 @@
+package ro.msg.learning.shop.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.dto.ProductCategoryDTO;
+import ro.msg.learning.shop.dto.builders.ProductCategoryBuilder;
+import ro.msg.learning.shop.model.Category;
+import ro.msg.learning.shop.model.Product;
+import ro.msg.learning.shop.model.Supplier;
+import ro.msg.learning.shop.service.CategoryService;
+import ro.msg.learning.shop.service.ProductService;
+import ro.msg.learning.shop.service.SupplierService;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/productCategory")
+public class CategoryController {
+    private final ProductService productService;
+    private final CategoryService productCategoryService;
+    private final SupplierService supplierService;
+
+    @PostMapping("/create")
```

Comment:
For the basic CRUD operations within a REST API there are some standards that should be respected. With only 2 paths and the request type (get, post, ...) you can configure almost 10 endpoints. Here is a photo which should make it clear how the endpoints should look like.
![image](https://user-images.githubusercontent.com/38504534/183597315-22f00dbf-2a53-4cbf-a4bb-d04fc017653e.png)